### PR TITLE
Added traceparent header to all HTTP payloads

### PR DESCRIPTION
**Pull request by robert-smartbear**
## Goal

Add trace id to all HTTP payloads

## Design

bugsnag-flutter-performance provides a callback to bugsnag-flutter-bridge that creates traceparent header with the current trace ID and span ID. Then, other packages such as bugsnag-flutter-dart-io-http-client and bugsnag-flutter-http-client call it and add the header to their HTTP payload

### CODE REVIEW

File: packages/bugsnag_flutter_performance/lib/src/client.dart [hunk 1]
```diff
@@ -130,6 +131,7 @@ class BugsnagPerformanceClientImpl implements BugsnagPerformanceClient {
         networkRequestCallback,
     String? releaseStage,
     List<String>? enabledReleaseStages,
+    List<String>? tracePropagationUrls,
```
Comment (lemnik) [inline packages/bugsnag_flutter_performance/lib/src/client.dart:134]: Given that these are matches as regex's I think this list should make that clearer:
```suggestion
    List<RegExp>? tracePropagationUrls,
```

**Review by richardelms — COMMENTED** [review]
Will tests be added to this PR or will they come at a later date? In which case should we use an integration branch?

**Comment by robert-smartbear** [comment]
> Will tests be added to this PR or will they come at a later date? In which case should we use an integration branch?

Added E2E tests

**Review by richardelms — APPROVED** [review]
LGTM pending passing CI run 👍

File: packages/bugsnag_flutter_performance/lib/src/client.dart [hunk 2]
```diff
@@ -453,6 +459,45 @@ class BugsnagPerformanceClientImpl implements BugsnagPerformanceClient {
...
+    return result;
+  }
+
+  bool _shouldAddTraceHeader(String url) {
+    final tracePropagationUrls = configuration?.tracePropagationUrls;
+    if (tracePropagationUrls != null && tracePropagationUrls.isNotEmpty) {
+      return tracePropagationUrls.indexWhere((regex) {
+            return regex.hasMatch(url);
+          }) !=
+          -1;
```
Comment (lemnik) [inline packages/bugsnag_flutter_performance/lib/src/client.dart:488]: nit:

```suggestion
      return tracePropagationUrls.any((regex) => regex.hasMatch(url));
```

**Review by lemnik — APPROVED** [review]
LGTM, with one small suggestion

**Pull request merged**